benches/lua.rs

```diff
@@ -9,167 +8,121 @@ use vector::{
     Event,
 };
 
-fn add_fields(c: &mut Criterion) {
+fn bench_add_fields(c: &mut Criterion) {
+    let event = Event::new_empty_log();
+
     let key = "the key";
     let value = "this is the value";
 
-    let value_bytes_native = Bytes::from(value).into();
-    let value_bytes_v1 = Bytes::from(value).into();
-    let value_bytes_v2 = Bytes::from(value).into();
+    let mut group = c.benchmark_group("lua_add_fields");
+    group.throughput(Throughput::Elements(1));
 
-    c.bench(
-        "lua_add_fields",
-        Benchmark::new("native", move |b| {
-            b.iter_with_setup(
-                || {
-                    let mut map = IndexMap::new();
-                    map.insert(String::from(key), String::from(value).into());
-                    (
-                        Event::new_empty_log(),
-                        transforms::add_fields::AddFields::new(map, true).unwrap(),
-                    )
-                },
-                |(event, mut transform)| {
-                    let event = transform.transform(event).unwrap();
-                    assert_eq!(event.as_log()[key], value_bytes_native);
-                },
-            )
-        })
-        .with_function("v1", move |b| {
-            b.iter_with_setup(
-                || {
-                    let source = format!("event['{}'] = '{}'", key, value);
-                    (
-                        Event::new_empty_log(),
-                        transforms::lua::v1::Lua::new(&source, vec![]).unwrap(),
-                    )
-                },
-                |(event, mut transform)| {
-                    let event = transform.transform(event).unwrap();
-                    assert_eq!(event.as_log()[key], value_bytes_v1);
-                },
-            )
-        })
-        .with_function("v2", move |b| {
-            b.iter_with_setup(
-                || {
-                    let config = format!(
-                        r#"
-                        hooks.process = """
-                            function (event, emit)
-                                event.log['{}'] = '{}'
+    let mut benchmarks = [
```

Comment:
The body of the iter loop for each benchmark was the same so I refactored this to be an array of (name, transform) tuples. I do get the feeling that this could be a little overboard though :smile: .

Reply:
I like it! You might prefer to use `AddFields::from(toml::from_str("my config block"))` for consistency's sake.